generators/beamAndVertexGenerator.h

```diff
@@ -61,22 +80,48 @@ namespace rpwa {
 		bool _readBeamfileSequentially;
 		long _currentBeamfileEntry;
 
+		virtual bool loadBeamFileWithMomenta();
+		virtual bool loadBeamFileWithInclinations();
+		virtual bool eventSimple(const rpwa::Target& target, const rpwa::Beam& beam);
+		virtual bool eventFromMomenta(const rpwa::Target& target, const rpwa::Beam& beam);
+		virtual bool eventFromMomentaCorrectMomentumResolution(const rpwa::Target& target, const rpwa::Beam& beam);
+		virtual bool eventFromInclinations(const rpwa::Target& target, const rpwa::Beam& beam);
+		/**
+		 * Load next event from beam file into the member variables of this object
+		 * @return true if loading was successful
+		 */
+		virtual bool loadNextEventFromBeamfile();
+
 	  private:
 
-		bool _simpleSimulation;
+		simulationMode _simulationMode;
 
 		TFile* _rootFile;
 		TTree* _beamTree;
 
-		// pairs with [value, sigma]
-		std::pair<double, double> _vertexX;
-		std::pair<double, double> _vertexY;
-		std::pair<double, double> _beamMomentumX;
-		std::pair<double, double> _beamMomentumY;
-		std::pair<double, double> _beamMomentumZ;
+		double _vertexX;
```

Comment:
It would be nice to add brief comments what the variables mean/hold; maybe also group according to different tree formats